internal/datasource/common/attributesources/attribute_sources_state.go

```diff
@@ -0,0 +1,167 @@
+package attributesources
```

Comment:
Is this necessary? It doesn't seem to be using any datasource-specific logic, so can we just use the existing common logic for attribute_sources?

Generally the schema types for data sources seem to be different, but outside of schema I'm not sure this is necessary.